tests/integration/generic_test.go

```diff
@@ -52,6 +52,22 @@ var _ = Describe("odo generic", func() {
 		Expect(output).To(ContainSubstring("Invalid command - see available commands/subcommands above"))
 	})
 
+	It("returns JSON error when using an invalid command with JSON output", func() {
+		res := helper.Cmd("odo", "unknown-command", "-o", "json").ShouldFail()
+		stdout, stderr := res.Out(), res.Err()
+		Expect(stdout).To(BeEmpty())
+		Expect(helper.IsJSON(stderr)).To(BeTrue())
+
+	})
+
+	It("returns JSON error when using an invalid sub-command with JSON output", func() {
+		res := helper.Cmd("odo", "describe", "unknown-sub-command", "-o", "json").ShouldFail()
+		stdout, stderr := res.Out(), res.Err()
+		Expect(stdout).To(BeEmpty())
+		Expect(helper.IsJSON(stderr)).To(BeTrue())
+
+	})
```

Comment:
Perhaps you can combine these error messages in a single It spec to reduce setup overhead?
There's one more case of `odo describe -o json` where `odo describe` cannot function as a standalone command.